tests/autoyast/console.pm

```diff
@@ -27,6 +27,7 @@ sub run {
     my ($self) = @_;
     $self->wait_boot if check_var('BACKEND', 'ipmi');
     select_console 'root-console';
+    assert_screen("login-console-motd-message") if get_var("SALT_FORMULAS_PATH");
```

Comment:
I think we better name it `login-console-motd-message-changed` so if someone create needle in future will expect the difference, if not, it would lead to some false positive, better would be not to check it with needle but not sure if that is feasible...

Reply:
here it is only validate the motd after provision. not the change.

Reply:
It is validating the changed one by the salt form, the one that is provisioned in the system and not the other one from the tarball, so what happen if we got the text from the tarball, it would not be any detection of a bug in such a case...
Correct me if I'm wrong, but default is `Have a lot of fun...` right? So that could be another false positive I think.
So if you name the needle with something that indicates that is not expecting a default message it would be helpful for the ones creating needle.
Another thing besides the needle that would be more important would be to check `cat /etc/motd` to avoid any ambiguity about what is expected, including test data for it, instead of hardcoding it when `type_string "Happy Testing changed";`

Reply:
As an alternative to avoid extra needles, we could just validate content of `/etc/motd`. With this change we are mixing multiple things in single module. I would suggest moving validation to separate module so it's clear where validation actually happens.

Reply:
> It is validating the changed one by the salt form, the one that is provisioned in the system and not the other one from the tarball, so what happen if we got the text from the tarball, it would not be any detection of a bug in such a case...

Thats correct as assumption. And in this case the test fails as the needle will not match. 

> Correct me if I'm wrong, but default is Have a lot of fun... right? So that could be another false positive I think.

Default is Happy Testing because i have changed it to something to see the reflection

> Another thing besides the needle that would be more important would be to check cat /etc/motd to avoid any ambiguity about what is expected, including test data for it, instead of hardcoding it when type_string "Happy Testing changed";

Thanks for it. i will try it when i found time.

> As an alternative to avoid extra needles, we could just validate content of /etc/motd. With this change we are mixing multiple things in single module. I would suggest moving validation to separate module so it's clear where validation actually happens.

I thought of a separate module but i was like this place makes totally sense to be used. I try to think what it is more beneficial. i guess your suggestion is more clear. let me see

Reply:
so i went in the opinion of a separate module and avoiding needling.